config/rbac/gc_clusterrole.yaml

```
@@ -0,0 +1,32 @@
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: gc
```

Should we use `nfd-` prefix for all the names. I think this is relevant especially for the cluster-scoped resources like this.

```suggestion
  name: nfd-gc
```

---

when `make deploy` runs kustomize it appends `nfd-` to all the names in the config it generates  (set by the `namePrefix: nfd-` setting in config/default/kustomization.yaml), so all the roles, service-accoutns etc should be named nfd-*

---

Hmm, ok. I still find it confusing that the clusterrole references a ServiceAccount with a different name than what's here 🤨 If the names are all replaced could we just use a placeholder (like `NAME`) for all of them to make it clear that it will be replaced. Or is it too obvious (to everyone else than me)?

---

maybe we might remove the namePrefix? IMHO the definitions of the roles/accounts/etc' become more readable, we don't need to take all the substitutions into account when reading the yamls..

---

> IMHO the definitions of the roles/accounts/etc' become more readable, we don't need to take all the substitutions into account when reading the yamls..

I agree. Would make it more readable and easier to review and spot mistakes

---

I agree, it is a bit confusing as is, while testing it took me a bit to figure out why I was getting names like `nfd-nfd-gc`   I left it in to try and keep the focus of the PR tight, however it sounds like we all agree that its worth doing, so I will make the change and update the PR.

---

> I left it in to try and keep the focus of the PR tight,

Mm, that could also be done in a separate (prior) PR. If you like.